spanner/api/Program.cs

```diff
@@ -0,0 +1,989 @@
+                $"Data Source=projects/{projectId}/instances/"
+                + $"{instanceId}/databases/{databaseId}";
+            string createStatement =
+                $"CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) "
+                + "STORING (MarketingBudget)";
+            // Make the request.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var createCmd = connection.CreateDdlCommand(createStatement);
+                await createCmd.ExecuteNonQueryAsync();
+            }
+            return;
+            // [END create_storing_index]
+        }
+
+        public static async Task AddColumnAsync(
+    string projectId, string instanceId, string databaseId)
+        {
+            // [START add_column]
+            // Initialize request argument(s).
+            string connectionString =
+                $"Data Source=projects/{projectId}/instances/"
+                + $"{instanceId}/databases/{databaseId}";
+            string createStatement =
+                $"ALTER TABLE Albums ADD COLUMN MarketingBudget INT64";
+            // Make the request.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var createCmd = connection.CreateDdlCommand(createStatement);
+                await createCmd.ExecuteNonQueryAsync();
+            }
+            Console.WriteLine("Added the MarketingBudget column.");
+            return;
+            // [END add_column]
+        }
+
+        public static async Task QuerySampleDataAsync(
+            string projectId, string instanceId, string databaseId)
+        {
+            // [START query_data]
+            string connectionString =
+            $"Data Source=projects/{projectId}/instances/"
+            + $"{instanceId}/databases/{databaseId}";
+            // Create connection to Cloud Spanner.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                var cmd = connection.CreateSelectCommand(
+                    "SELECT SingerId, AlbumId, AlbumTitle FROM Albums");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        Console.WriteLine($"SingerId : {reader["SingerId"]} "
+                        + $"AlbumId : {reader["AlbumId"]} "
+                        + $"AlbumTitle : {reader["AlbumTitle"]}");
+                    }
+                }
+            }
+            return;
+            // [END query_data]
+        }
+
+
+        public static async Task QueryDataWithIndexAsync(
+            string projectId, string instanceId, string databaseId,
+            string startTitle = "Aardvark", string endTitle = "Goo")
+        {
+            Console.WriteLine("Running QueryDataWithIndex...");
+            // [START query_data_with_index]
+            // [START read_data_with_index]

```

Comment:
I think he is using this sample for both query and read since c# only supports query.